### PR TITLE
Add a Sequel health check

### PR DESCRIPTION
**Pull request by warrenseen**
We're using Sequel in a few projects now with Event Sourcery. Let's add a generic Sequel connection check that we can parameterise for our ES databases.

### CODE REVIEW

File: lib/rack/ecg/check/sequel_connection.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        def result
+          value = ""
+          status = "ok"
+          begin
+            if connection_parameters.nil?
+              status = "error"
+              value = "Sequel Connection parameters not found"
+            elsif defined?(Sequel)
+              ::Sequel.connect(connection_parameters) { |db|
+                value = db.valid_connection?
```
Comment (warrenseen) [inline lib/rack/ecg/check/sequel_connection.rb]: This call is missing a param. fixing...

File: lib/rack/ecg/check/sequel_connection.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+module Rack
+  class ECG
+    module Check
+      class SequelConnection
+        attr_reader :connection_parameters, :name
+        def initialize(parameters = {})
+          puts parameters
```
Comment (denmat) [inline lib/rack/ecg/check/sequel_connection.rb]: debug code?

File: lib/rack/ecg.rb [hunk 3]
```diff
@@ -52,10 +52,10 @@ def call(env)
     def build_check_classes(check_names)
       check_names = Array(check_names) # handle nil, or not a list
       check_names = check_names | DEFAULT_CHECKS # add the :http check if it's not there
-      check_names.map{|check_name|
+      check_names.map { |check_name, check_params|
         check_class = CheckRegistry.instance[check_name]
         raise "Don't know about check #{check_name}" unless check_class
-        check_class
+        [check_class, check_params]
```
Comment (petervandoros) [inline lib/rack/ecg.rb]: Can your write some tests around this change in behaviour?

File: lib/rack/ecg/check/sequel_connection.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
+module Rack
+  class ECG
+    module Check
+      class SequelConnection
```
Comment (petervandoros) [inline lib/rack/ecg/check/sequel_connection.rb]: Would be nice to add something to the docs around how to use this check.

File: lib/rack/ecg.rb (hunk 3 above)
Comment (warrenseen) [inline lib/rack/ecg.rb]: @petervandoros Yeah, I can, although the existing spec already exercises this pathway anyway

File: lib/rack/ecg/check/sequel_connection.rb (hunk 4 above)
Comment (warrenseen) [inline lib/rack/ecg/check/sequel_connection.rb]: Yep, will do, wanted to get some early feedback, hence the fact that this is WIP 😄

File: lib/rack/ecg.rb (hunk 3 above)
Comment (petervandoros) [inline lib/rack/ecg.rb]: @warrenseen 
I say it more from the point of view in making check params first class citizens and being explicitly tested as opposed to the code path being exercised as part of other tests.

File: lib/rack/ecg/check/sequel_connection.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        end
+
+        def result
+          value = ""
+          status = "ok"
+          begin
+            if connection_parameters.nil?
+              status = "error"
+              value = "Sequel Connection parameters not found"
+            elsif defined?(Sequel)
```
Comment (twe4ked) [inline lib/rack/ecg/check/sequel_connection.rb]: Should this be checking `::Sequel`?

**Pull request merged**